FreeRTOS-Plus/Demo/FreeRTOS-IoT-Libraries-LTS-Beta2/mqtt/mqtt_light_weight/DemoTasks/LightWeightMQTTExample.c

```diff
@@ -41,41 +54,938 @@
+ *
+ * @param pxIncomingPacket is a pointer to structure containing deserialized
+ * MQTT response.
+ * @param usPacketId is the packet identifier from the ack received.
+ */
+static void prvMQTTProcessResponse( MQTTPacketInfo_t * pxIncomingPacket,
+                                    uint16_t usPacketId );
+
+/**
+ * @brief Process incoming Publish message.
+ *
+ * @param pxPublishInfo is a pointer to structure containing deserialized
+ * Publish message.
+ */
+static void prvMQTTProcessIncomingPublish( MQTTPublishInfo_t * pxPublishInfo );
+
+/**
+ * @brief Receive and validate MQTT packet from the broker, determine the type
+ * of the packet and processes the packet based on the type.
+ *
+ * @param xMQTTSocket is a TCP socket that is connected to an MQTT broker to which
+ * an MQTT connection has been established.
+ */
+static void prvMQTTProcessIncomingPacket( Socket_t xMQTTSocket );
+
+/**
+ * @brief The transport receive wrapper function supplied to the MQTT library for
+ * receiving type and length of an incoming MQTT packet.
+ *
+ * @param[in] pxContext Pointer to network context.
+ * @param[out] pBuffer Buffer for receiving data.
+ * @param[in] bytesToRecv Size of pBuffer.
+ *
+ * @return Number of bytes received or zero to indicate transportTimeout;
+ * negative value on error.
+ */
+static int32_t prvTransportRecv( NetworkContext_t * pxContext,
+                                 void * pvBuffer,
+                                 size_t xBytesToRecv );
+
+/**
+ * @brief Generate and return monotonically increasing packet identifier.
+ *
+ * @return The next PacketId.
+ *
+ * @note This function is not thread safe.
+ */
+static uint16_t prvGetNextPacketIdentifier( void );
+
+/*-----------------------------------------------------------*/
+
+/* @brief Static buffer used to hold MQTT messages being sent and received. */
+static uint8_t ucSharedBuffer[ mqttexampleSHARED_BUFFER_SIZE ];
+
+/**
+ * @brief Packet Identifier generated when Subscribe request was sent to the broker;
+ * it is used to match received Subscribe ACK to the transmitted subscribe request.
+ */
+static uint16_t usSubscribePacketIdentifier;
+
+/**
+ * @brief Packet Identifier generated when Unsubscribe request was sent to the broker;
+ * it is used to match received Unsubscribe response to the transmitted unsubscribe
+ * request.
+ */
+static uint16_t usUnsubscribePacketIdentifier;
+
+
+/** @brief Static buffer used to hold MQTT messages being sent and received. */
+const static MQTTFixedBuffer_t xBuffer =
+{
+    .pBuffer = ucSharedBuffer,
+    .size    = mqttexampleSHARED_BUFFER_SIZE
+};
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief The Network Context implementation. This context is passed to the
+ * transport interface functions.
+ *
+ * This example uses transport interface function only to read the packet type
+ * and length of the incoming packet from network.
+ */
+struct NetworkContext
```

Comment:
Should this be  typedef to NetworkContext_t ?

As per other other comment we can simplify this like this :
typedef Socket_t NetworkContext_t

Reply:
No, the typedef is in transport_interface.h: https://github.com/FreeRTOS/FreeRTOS/blob/lts-development/FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/c_sdk/platform/transport_interface.h#L34